### DISCUSSION TITLE
N98-Magerun needs your help

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#3585] [Extensions & Modules]
Most of you have used **N98-Magerun** with Magento/OpenMage and if you haven't, you have heard of this "must-have" tool that was a source of inspiration for the Magento 2 team, when introduced the feature to run administration commands in the CLI.

Actions that can be performed in the CLI were recently discussed, such as changing the email address in an order, associating an order placed by a guest to a registered customer, undo of canceled orders and many others. I think that **N98-Magerun** can be extended with plugins to solve these objectives. For those who use DDEV as a test environment, it comes as a help, because solutions for expanding the commands were discussed already, some of them being integrated.

**N98-Magerun** needs our help to successfully pass the testing phase and make it compatible with the new versions > PHP 8.1. The hard work was done by @cmuench and @sreichel, whom I would like to thank for not letting this wonderful project die for those who still love the Magento 1 platform. 

**Details here**
https://github.com/netz98/n98-magerun/issues/1275

**Comment by sreichel** [comment]
I highly appreciate any help for testing.

Atm I'm refactoring most commands, remove M2 and EE related code, prepare for symfony 6.4 (for next release?). 

I also want/have to check magerun-addons from e.g @kalenjordan @peterjaap and provide updates ...

Every little help would be nice.

If you have ideas for new (small) commands or improvements plmk.

@ADDISON74 thanks!

**Reply by addison74** [reply]
Related to **Orders**, the following would be useful

1 - **Changing the email address associated with an order.** If it was filled in incorrectly by the customer in the order form, we are not able to send email updates to the customer from the Backend. As it is now the only way to change it is manually in the database.

2 - **Associating a guest order with a registered buyer.**

3 - **Open a closed order.** There are situations when it is accidentally closed or the customer returns after a short period of time. Reopening helps the customer history.

**Reply by sreichel** [reply]
1. easy
2. should be covered by https://github.com/kalenjordan/magerun-addons#assign-order-to-new-customer (with some limitations - same as PR here)
3. needs more information/investigation. (change state, status, waht else is needed?)

**Reply by addison74** [reply]
At point no. 3, the status changes, but the quantities must decrease again from the stock.

**Comment by sreichel** [comment]
@ADDISON74 please close this. magerun v3-dev should work with 8.1.

**Comment by sreichel** [comment]
Feel free to test https://github.com/netz98/n98-magerun/pull/1484
